homeassistant/components/knx/binary_sensor.py

```diff
@@ -34,11 +34,11 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the KNX binary sensor platform."""
-    xknx: XKNX = hass.data[DOMAIN].xknx
+    knx_module: KNXModule = hass.data[DOMAIN]
```

Comment:
For a future improvement; this could be moved to the runtime_data property of the config entry (removing the need to use hass.data overall)

Reply:
Unfortunately there are some places in the integration where I do need it, but don't have a config_entry handle.
Device triggers
https://github.com/home-assistant/core/blob/dev/homeassistant/components/knx/device_trigger.py#L66-L70
Services
https://github.com/home-assistant/core/blob/dev/homeassistant/components/knx/services.py#L225-L227
Websockets
https://github.com/home-assistant/core/blob/dev/homeassistant/components/knx/websocket.py#L122-L128

For Websockets we could probably pass an instance before registering. 
But services shall be registered in `async_setup()` so that is too early and for device triggers I don't know either.